pkg/controllers/provisioning/scheduling/instance_selection_test.go

```diff
@@ -603,4 +644,932 @@ var _ = Describe("Instance Type Selection", func() {
+				Resources: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("1"),
+					v1.ResourceMemory: resource.MustParse("1Gi"),
+				},
+			}
+			opts1.Offerings = []cloudprovider.Offering{
+				{
+					CapacityType: v1beta1.CapacityTypeSpot,
+					Zone:         "test-zone-1-spot",
+					Price:        0.52,
+					Available:    true,
+				},
+			}
+			opts2 := fake.InstanceTypeOptions{
+				Name:             "instance-type-2",
+				Architecture:     v1beta1.ArchitectureArm64,
+				OperatingSystems: sets.New(string(v1.Linux)),
+				Resources: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("4"),
+					v1.ResourceMemory: resource.MustParse("4Gi"),
+				},
+			}
+			opts2.Offerings = []cloudprovider.Offering{
+				{
+					CapacityType: v1beta1.CapacityTypeSpot,
+					Zone:         "test-zone-1-spot",
+					Price:        1.0,
+					Available:    true,
+				},
+			}
+			instanceTypes = append(instanceTypes, fake.NewInstanceType(opts1))
+			instanceTypes = append(instanceTypes, fake.NewInstanceType(opts2))
+			cloudProvider.InstanceTypes = instanceTypes
+
+			// Define NodePool that has minValues on instance-type requirement.
+			nodePool.Spec.Template.Spec.Requirements = []v1beta1.NodeSelectorRequirementWithMinValues{
+				{
+					NodeSelectorRequirement: v1.NodeSelectorRequirement{
+						Key:      v1.LabelInstanceTypeStable,
+						Operator: v1.NodeSelectorOpIn,
+						Values:   []string{"instance-type-1", "instance-type-2"},
+					},
+					MinValues: lo.ToPtr(2),
+				},
+			}
+			ExpectApplied(ctx, env.Client, nodePool)
+
+			// 2 pods are created with resources such that both fit together only in one of the 2 InstanceTypes created above.
+			// Both of these should schedule on a instance-type-2 without the minValues requirement being specified.
+			pod1 := test.UnschedulablePod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("0.9"),
+					v1.ResourceMemory: resource.MustParse("0.9Gi")},
+				},
+			})
+			pod2 := test.UnschedulablePod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("0.9"),
+					v1.ResourceMemory: resource.MustParse("0.9Gi")},
+				},
+			})
+
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod1, pod2)
+			node1 := ExpectScheduled(ctx, env.Client, pod1)
+			node2 := ExpectScheduled(ctx, env.Client, pod2)
+
+			// This ensures that the pods are scheduled in 2 different nodes.
+			Expect(node1.Name).ToNot(Equal(node2.Name))
+
+			// Ensures that NodeClaims are created with 2 instanceTypes
+			Expect(len(supportedInstanceTypes(cloudProvider.CreateCalls[0]))).To(BeNumerically(">=", 2))
+		})
+		It("should schedule respecting the minValues in Gt operator.", func() {
+			// custom key that will help us with numerical values to be used for Gt operator
+			instanceGeneration := "karpenter/numerical-value"
```

Comment:
```suggestion
			numericalRequirementKey := v1beta1.Group + "/numerical-value"
```

nit: Typically you prefix with the group name

Reply:
It doesn't allow you do attach the label with `karpenter.sh` as it complains that it is a restricted label like I said [here](https://github.com/nikmohan123/karpenter/pull/1#discussion_r1493098381).

Error: 
```
got an error:
      <*errors.StatusError | 0x1400069a1e0>:
      NodePool.karpenter.sh "speakerleather-1-znthvad9w8" is invalid: spec.template.spec.requirements[0].key: Invalid value: "string": label domain "karpenter.sh" is restricted
      {
```

So, not fixing this and for the rest.